### PR TITLE
(#1501796) backport of unit_watch_pid() with new 'exclusive' flag (PR #12033)

### PR DESCRIPTION
**Pull request by dtardon**
Signed-off-by: Renaud Métrich <rmetrich@redhat.com>

Resolves: #1501796

### CODE REVIEW

File: src/core/unit.c [hunk 1]
```diff
@@ -1985,7 +1995,12 @@ int unit_watch_pid(Unit *u, pid_t pid) {
                         return r;
 
                 r = hashmap_put(u->manager->watch_pids2, LONG_TO_PTR(pid), u);
-        }
+                if (r >= 0)
+                        log_unit_debug(u->id, "Watching "PID_FMT" through watch_pids2.", pid);
+                else if (r == -EEXIST)
+                        log_unit_debug(u->id, "Warning: cannot watch "PID_FMT", PID is already watched twice.", pid);
```
[the remark this answers is not in the document]
Comment (msekletar) [reply src/core/unit.c]: I think that is a good idea. In an ideal world, this situation shouldn't really happen very often anyway.

[the remark this answers is not in the document]
Comment (rmetrich) [reply src/core/unit.c]: @dtardon Hello, as discussed yesterday with @msekletar please change line 2001 from **log_unit_debug** to **log_unit_warning** and remove `Warning:` prefix and we are good to merge.

[the remark this answers is not in the document]
Comment (dtardon) [reply src/core/unit.c]: Done.

**Review by msekletar — CHANGES_REQUESTED** [review]
David, can you please also drop the Signed-off line and provide a reference to upstream commit?

**Comment by msekletar** [comment]
Actually we should properly cherry-pick the upstream commit also with the commit message.

**Comment by dtardon** [comment]
Updated.

**Review by msekletar — APPROVED** [review]
LGTM

**Pull request merged**